data/modules/DonateToCranks/DonateToCranks.lua

```diff
@@ -33,46 +34,65 @@ local addReputation = function (money)
 	else
 		newRep = curRep + 2^(math.log(money)/math.log(10))
 	end
-	Character.persistent.player.reputation = newRep
+	char.reputation = newRep
+	return newRep
 end
 
 
-local onChat = function (form, ref, option)
-	local ad = ads[ref]
+local computeReputation = function (ad)
+	-- compute money to get to next level of reputation
 
-	if option == 0 then
-		form:Clear()
+	local current_reputation = 	Character.persistent.player:GetReputationRating()
+	print("\n\nCURRENT:", current_reputation)
 
-		form:SetTitle(ad.title)
-		form:SetFace({ seed = ad.faceseed })
-		form:SetMessage(ad.message)
+	for i=0,5,1 do
+		local donate = math.floor(10^i)
+		local clone = Character.persistent.player:Clone()
+		addReputation(donate * ad.modifier, clone)
+		local new_reputation = clone:GetReputationRating()
+		print("money, new_rep_score, new_rep:", donate, clone.reputation, new_reputation)
+		if new_reputation ~= current_reputation then
+			return donate
+		end
+	end
+	-- aught not come here unless player has maxed out reputation
+	return -1
+end
 
-		form:AddOption(Format.Money(1,false), 1)
-		form:AddOption(Format.Money(10,false), 10)
-		form:AddOption(Format.Money(100,false), 100)
-		form:AddOption(Format.Money(1000,false), 1000)
-		form:AddOption(Format.Money(10000,false), 10000)
-		form:AddOption(Format.Money(100000,false), 100000)
 
-		return
-	end
+local onChat = function (form, ref, option)
+	local ad = ads[ref]
+	form:Clear()
+
+	form:SetTitle(ad.title)
+	form:SetFace(ad.charcter)
```

Comment:
`charcter` -> `character`.
I would prefer to restore deterministic seeds for face generation, but for now this is an acceptable compromise.

Reply:
Yes, me too, especially since the Character class documents the functionality, I think. Couldn't figure it out though. Not sure what the best route would be? Overloading the `ChatForm:SetFace(character)` to also take a table with other variables? If you have time at some point we can discuss it